activejob/lib/active_job/queue_adapters/test_adapter.rb

```diff
@@ -55,9 +55,11 @@ def perform_or_enqueue(perform, job, job_data)
 
         def filtered?(job)
           if filter
-            !Array(filter).include?(job.class)
+            !Array(filter).include?(job.class) || (queue && job.queue_name != queue.to_s)
           elsif reject
-            Array(reject).include?(job.class)
+            Array(reject).include?(job.class) || (queue && job.queue_name != queue.to_s)
+          elsif queue
+            job.queue_name != queue.to_s
           else
             false
           end
```

Comment:
Can hoist up the `queue` conditional since it applies to every clause here:
```ruby
def filtered?(job)
  filtered_queue?(job) || filtered_job_class?(job)
end

def filtered_queue?(job)
  if queue
    job.queue_name != queue.to_s
  end
end

def filtered_job_class?(job)
  if filter
    !Array(filter).include?(job.class)
  elsif reject
    Array(reject).include?(job.class)
  end
end
```

Reply:
:heart_eyes: Thanks for the tip. I like this simplified format. I applied this suggestion in https://github.com/rails/rails/commit/ec2e8f645eded183224420e209420376c63b99bb and added you to authors of this commit.